golem-rib-repl/src/rib_context.rs

```diff
@@ -0,0 +1,36 @@
+use crate::{RawRibScript, ReplPrinter};
+use rib::RibCompiler;
+
+// A projection of internal repl_state that could be useful
+// for advanced customisation of REPL commands.
+pub struct ReplContext<'a> {
+    printer: &'a dyn ReplPrinter,
+    rib_script: &'a RawRibScript,
+    compiler: &'a RibCompiler,
```

Comment:
Example: compiler is always retrieved from internal state (because a compiler of Rib can change if there is a dynamic load of a new component) and made it available for any clients who want to build a new component, but they will be restricted from making any change or mutate it.